prow/config/config.go

```diff
@@ -1498,6 +1498,23 @@ type SlackReporter struct {
 // Use `org/repo`, `org` or `*` as key and an `SlackReporter` struct as value.
 type SlackReporterConfigs map[string]SlackReporter
 
+func (cfg SlackReporterConfigs) mergeFrom(additional *SlackReporterConfigs) error {
```

Comment:
I believe this needs to be pointer receiver, otherwise the `cfg[orgOrRepo] = slackReport` has no effect. see https://go.dev/tour/methods/4

Reply:
And if the above statement is correct then there is at least one unit test missing here

Reply:
I think it applies for struct receivers. Maps and other similar types are to behave differently (and ``SlackReporterConfigs`` is essentially ``map[string]SlackReporter``):
https://go.dev/play/p/t7Tjy-hAlIk

I'll also make sure of the coverage, let me take a look to confirm it's there

Reply:
ack, good to know

Reply:
It does lack coverage, and the same goes for other mergable fields in tide plugin (specifically tide.merge-type) and branch-policy in branch-protection. I know it by removing similar lines of code and running ``make go-unit`` successfully :facepalm: 

I think most of the merging logic is being tested on the relevant openshift repository, but not in here
I don't mind taking care of all those tests (or just SlackReporterConfigs ones) in this PR, or in the following one. Whatever your preference is

Reply:
yeah feel free to unhold as you feel ready

Reply:
Ack. I'll do a follow-up PR covering more of the config-merging code
Thank you!